### PR TITLE
174a back end doc

### LINKED ISSUE #174: documentation back-end
### Content
To guaranty maintainability, all code should be documented as clear as possible.

### Feature
- documentation for the communication, config, handler

### Action plan
- [ ] configHandler.go
- [ ] readConfigTypes.go
- [ ] workingConfigTypes.go
- [ ] handler.go
- [ ] handler_helper.go
- [ ] communication.go

### Related issues
GH-173

### PR DESCRIPTION
**Pull request by mrlomar**
## Contents
All methods in the backend now have documentation

## Issue
Closes #174

### CODE REVIEW

File: back-end/src/sciler/config/configHandler.go [hunk 1]
```diff
@@ -202,17 +212,27 @@ func checkActions(actions []Action, config WorkingConfig) []string {
 	return errorList
 }
 
+// checkActionTimer is a method that checks the current config for mistakes in the action of a timer
```
[the remark this answers is not in the document]
Comment (mrlomar) [reply back-end/src/sciler/config/configHandler.go]: done

File: back-end/src/sciler/handler/handler_helper.go [hunk 2]
```diff
@@ -33,12 +35,21 @@ func (handler *Handler) SendSetup() {
...
+		// its return message will be handled in a new pipeline
+		// This way, the front-end will receive all device information from the config,
+		// and after that the current status from all the connected devices
 		handler.GetStatus(value.ID)
 	}
 	handler.sendEventStatus()
 }
 
 // SendComponentInstruction sends a list of instructions to a client, with a delay if given a valid duration.
+// This function is called in reaction to a execution of a rule
```
[the remark this answers is not in the document]
Comment (mrlomar) [reply back-end/src/sciler/handler/handler_helper.go]: done

File: back-end/src/sciler/handler/handler_helper.go [hunk 3]
```diff
@@ -49,17 +60,26 @@ func (handler *Handler) SendComponentInstruction(clientID string, instructions [
...
+			handler.Communicator.Publish(clientID, string(jsonMessage), 3)
+		}()
 	} else {
 		logger.Infof("sending instruction data to %s: %s", clientID, fmt.Sprint(message.Contents))
 		handler.Communicator.Publish(clientID, string(jsonMessage), 3)
 	}
 }
 
 // SendLabelInstruction provides the action with a componentID from de LabelMap and a device to send it to
+// This function is called in reaction to a execution of a rule
```
[the remark this answers is not in the document]
Comment (mrlomar) [reply back-end/src/sciler/handler/handler_helper.go]: done

File: back-end/src/sciler/config/configHandler.go [hunk 4]
```diff
@@ -437,7 +454,9 @@ func generateButtonEvents(buttonEvents []ReadRule, config *WorkingConfig) (map[s
 	return newEvents, errorList
 }
 
-// generateRules creates a list rule objects with properly checked inner values
+// generateRules transforms readRules to rules
+// it generates conditions and copies the rest
+// if the config does not follow the manual, a non-empty list of mistakes is returned>>>>>>> develop-sprint-7
```
Comment (IssaHanou) [inline back-end/src/sciler/config/configHandler.go]: merge error

**Pull request merged**